quickwit/quickwit-metastore/src/metastore/postgres/metastore.rs

```diff
@@ -1329,40 +1331,77 @@ impl MetastoreService for PostgresqlMetastore {
         Ok(response)
     }
 
-    // TODO: Issue a single SQL query.
     async fn list_shards(
         &mut self,
         request: ListShardsRequest,
     ) -> MetastoreResult<ListShardsResponse> {
-        const LIST_SHARDS_QUERY: &str = include_str!("queries/shards/list.sql");
+        if request.subrequests.is_empty() {
+            return Ok(Default::default());
+        }
+        let mut sql_query_builder = Query::select();
 
-        let mut subresponses = Vec::with_capacity(request.subrequests.len());
+        for (idx, subrequest) in request.subrequests.iter().enumerate() {
+            let mut sql_subquery_builder = Query::select();
 
-        for subrequest in request.subrequests {
-            let shard_state: Option<&'static str> = match subrequest.shard_state() {
-                ShardState::Unspecified => None,
-                ShardState::Open => Some("open"),
-                ShardState::Closed => Some("closed"),
-                ShardState::Unavailable => Some("unavailable"),
-            };
-            let pg_shards: Vec<PgShard> = sqlx::query_as(LIST_SHARDS_QUERY)
-                .bind(subrequest.index_uid().to_string())
-                .bind(&subrequest.source_id)
-                .bind(shard_state)
-                .fetch_all(&self.connection_pool)
-                .await?;
+            sql_subquery_builder
+                .column(Asterisk)
+                .from(Shards::Table)
+                .and_where(Expr::col(Shards::IndexUid).eq(subrequest.index_uid().to_string()))
```

Comment:
do we still need the `.to_string()`?

Reply:
Yeah, `sea-query` uses a different `Value` type than `sqlx`, which only supports owned types.